doc/sphinx/reference/onedim/grid-refinement.md

```diff
@@ -0,0 +1,229 @@
+
+```{figure} /_static/images/onedim/refinement_large_right_interval.svg
+:width: 75%
+:alt: Grid with a spacing on the left interval that is too large.
+:align: center
+
+Grid with a spacing on the left interval that is too large.
+```
+
+the ratio, $ \alpha $ is calculated as:
+
+$$
+  \alpha = \frac{{z_{j+1} - z_j}}{{z_j - z_{j-1}}}
+$$
+
+
+These conditions are checked for all grid points in the domain, and if the ratio
+is greater than the user-specified value, then an additional grid point is inserted
+between the two points. To illustrate the interplay between the insertion of points
+in the domain and the marking of points to be kept, consider again the two cases above.
+
+For the case where the spacing on the right interval is too large, the grid is refined
+and a stencil of points around the point under consideration are marked to be kept.
+The figure below shows an example of the stencil of points that are marked to be kept
+when the spacing on the right interval is too large.
+
+```{figure} /_static/images/onedim/refinement_right_interval_stencil.svg
+:width: 75%
+:alt: The points that are marked as needing new grid points inserted to the right of
+them and which ones should be marked to be kept.
+:align: center
+
+The points that are marked as needing new grid points inserted to the right of them
+and which ones should be marked to be kept.
+```
+
+Likewise, for the case where the spacing on the left interval is too large, the grid is
+refined and a stencil of points around the point under consideration are marked to be kept.
+The figure below shows an example of the stencil of points that are marked to be kept
+when the spacing on the left interval is too large.
+
+```{figure} /_static/images/onedim/refinement_left_interval_stencil.svg
+:width: 75%
+:alt: The points that are marked as needing new grid points inserted to the right of
+them and which ones should be marked to be kept.
+:align: center
+
+The points that are marked as needing new grid points inserted to the right of them
+and which ones should be marked to be kept.
+```
+
+Marking the surrounding grid points prevents them from being removed during the refinement
+step that they were marked in. This is one of the ways that the algorithm ensures that
+the grid does not change too drastically during a refinement step.
+
+
+
+### Slope
+
+This refinement option controls the maximum change in a solution component between
+two adjacent grid points compared to the maximum change in that component across the
+entire domain. If the ratio of the change in a solution component between two adjacent
+grid points is greater than the user-specified value, then an additional grid point is
+inserted between the two points. This criterion is used to ensure that the solution
+is well-resolved in regions where the solution has steep gradients.
+
+The specific criteria that is being enforced is:
+$$
```

Comment:
You need a blank line before the opening `$$` for this to be rendered in display style. Please run Sphinx locally or download the output from the CI run and look at the HTML pages. There's really no other way to find all the formatting issues here.